plugins/map/EditingSupport.jsx

```diff
@@ -24,38 +26,23 @@ class EditingSupport extends React.Component {
         this.interaction = null;
         this.layer = null;
         this.currentFeature = null;
-        this.editStyle = [
-            new ol.style.Style({
-                fill: new ol.style.Fill({ color: [255, 0, 0, 0.5] }),
-                stroke: new ol.style.Stroke({ color: 'red', width: 2})
-            }),
-            new ol.style.Style({
-                image: new ol.style.RegularShape({
-                    fill: new ol.style.Fill({color: 'white'}),
-                    stroke: new ol.style.Stroke({color: 'red', width: 2}),
-                    points: 4,
-                    radius: 5,
-                    angle: Math.PI / 4
-                }),
-                geometry: (feature) => {
-                    if (feature.getGeometry().getType() === "Point") {
-                        return new ol.geom.MultiPoint([feature.getGeometry().getCoordinates()]);
-                    } else if (feature.getGeometry().getType() === "LineString") {
-                        return new ol.geom.MultiPoint(feature.getGeometry().getCoordinates());
-                    } else if (feature.getGeometry().getType() === "Polygon") {
-                        return new ol.geom.MultiPoint(feature.getGeometry().getCoordinates()[0]);
-                    } else if (feature.getGeometry().getType() === "MultiPoint") {
-                        return feature.getGeometry();
-                    } else if (feature.getGeometry().getType() === "MultiLineString") {
-                        return new ol.geom.MultiPoint(feature.getGeometry().getCoordinates()[0]);
-                    } else if (feature.getGeometry().getType() === "MultiPolygon") {
-                        return new ol.geom.MultiPoint(feature.getGeometry().getCoordinates()[0][0]);
-                    } else {
-                        return feature.getGeometry();
-                    }
-                }
-            })
-        ];
+        this.editStyle = FeatureStyles.interaction();
+        this.editStyle[1].setGeometry((feature) => {
```

Comment:
This is probably a bit hard to understand resp it requires you to know about the implementation detail of `FeatureStyles.interaction()`. It might be better to passs `geometry` as an argument to `FeatureStyles.interaction()` and then set the geometry accordingly inside `FeatureStyles.interaction()`.

Reply:
You're right, thanks. It's updated.
I can also split the first and the second style from "interaction" to be more clear if needed.

Reply:
Yes I think it would be better to split the interaction styles. Thanks.